olp-cpp-sdk-core/src/network2/curl/NetworkProtocolCurl.cpp

```diff
@@ -0,0 +1,1297 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "NetworkProtocolCurl.h"
+
+#include <algorithm>
```

Comment:
Some of these includes are already included in `NetworkProtocolCurl.h`. Please, double check them and remove excessive ones, like:
```
#include <memory>
#include <mutex>
#include <string>
#include <thread>

#include <curl/curl.h>
```